robot/rospackages/src/mcu_control/scripts/ArmNode.py

```diff
@@ -113,58 +98,148 @@ def stripFeedback(data):
+
+#    feedback_pub_topic = '/arm_feedback'
+#    rospy.loginfo('Beginning to publish to "'+feedback_pub_topic+'" topic')
+#    feedbackPub = rospy.Publisher(feedback_pub_topic, String, queue_size=10)
+
+#    subscribe_topic = '/arm_command'
+#    rospy.loginfo('Beginning to subscribe to "'+subscribe_topic+'" topic')
+#    #I could have a topic that listens for JointStateso and constructs the message here...
+#    sub = rospy.Subscriber(subscribe_topic, String, subscriber_callback)
+
+#    service_name = '/arm_request'
+#    rospy.loginfo('Waiting for "'+service_name+'" service request from client')
+#    serv = rospy.Service(service_name, ArmRequest, handle_client)
+
+#    # service requests are implicitly handled but only at the rate the node publishes at
+#    ser = get_serial()
+#    global reqFeedback
+#    reqFeedback=''
+#    global reqInWaiting
+#    reqInWaiting=False
+#    try:
+#        while not rospy.is_shutdown():
+#            #if I try reading from the serial port inside callbacks, bad things happen
+#            #instead I send the data elsewhere if required but only read from serial here.
+#            #not sure if I need the same precautions when writing but so far it seems ok.
+#            if ser.in_waiting:
+#                data=''
+#                feedback=None
+#                try:
+#                    data = ser.readline().decode()
+#                    feedback = stripFeedback(data)
+#                except:
+#                    rospy.logwarn('trouble reading from serial port')
+#                if feedback is not None:
+#                    if 'Motor Angles' in feedback:
+#                        #rospy.loginfo(feedback)
+#                        publish_joint_states(feedback)
+#                    elif 'battery voltage' in feedback:
+#                        left,voltage = feedback.split('battery voltage: ')
+#                        vBatPub.publish(float(voltage))
+#                    else:
+#                        #rospy.loginfo(feedback)
+#                        if reqInWaiting:
+#                            reqFeedback += feedback+'\r\n' #pass data to request handler
+#                        else:
+#                            if 'WARNING' in feedback:
+#                                rospy.logwarn(feedback)
+#                            #rospy.loginfo(feedback)
+#                            feedbackPub.publish(feedback)
+#                else:
+#                    rospy.loginfo('got raw data: '+data)
+#            rospy.Rate(100).sleep()
+#    except rospy.ROSInterruptException:
+#        pass
+
+#    def shutdown_hook():
+#        rospy.logwarn('This node ('+node_name+') is shutting down')
+#        ser.close() # good practice to close the serial port
+#        # do I need to clear the serial buffer too?
+#        time.sleep(1) # give ROS time to deal with the node closing (rosbridge especially)
+
+#    rospy.on_shutdown(shutdown_hook)
```

Comment:
saving in case it's needed?

Reply:
It will remind him that he has to send battery voltage, motor angles etc